### PR TITLE
[quant][refactor] Dedup the batch norm code

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #55761 [quant] ConvBNReLU1d
* #55760 [quant][refactor] Renaming layer names in the batch norm test
* #55759 [quant][refactor] BatchNorm factor out common code
* **#55758 [quant][refactor] Dedup the batch norm code**

Batch norm 2d and 3d were carbon copies of each other. This factors out the duplicated portions

Differential Revision: [D27702396](https://our.internmc.facebook.com/intern/diff/D27702396/)

### CODE REVIEW

File: torch/nn/quantized/modules/batchnorm.py [hunk 1]
```diff
@@ -1,28 +1,24 @@
 import torch
 import torch.nn.quantized.functional
-import torch.nn.intrinsic as nni
 
-class BatchNorm2d(torch.nn.BatchNorm2d):
-    r"""This is the quantized version of :class:`~torch.nn.BatchNorm2d`.
-    """
 
+class _BatchNormBase(torch.nn.modules.batchnorm._BatchNorm):
```
Comment (vkuzo) [inline torch/nn/quantized/modules/batchnorm.py:5]: optional: the conv classes use `_ConvNd` naming, we could also use `_BatchNormNd` here.  Or, use `_BatchNorm` to be consistent with the parent class.  Would be great to not introduce one additional naming scheme.

**Review by vkuzo — COMMENTED** [review]
logic lg, thanks for doing this!

What was the test plan for testing correctness?  Could we add that to the PR summary?  Otherwise it's hard to tell how this was actually tested.
